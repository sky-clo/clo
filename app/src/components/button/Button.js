import React from "react";

export default function Button(props) {
  const toolkitClass =
    props.primary === false ? "c-btn--secondary" : "c-btn--primary";
  const className = `c-btn ${toolkitClass} ${props.className}`;

  return (
    <button
      onClick={props.onClick}
      className={className}
      type={props.type}
      data-test={props.dataTest}
    >
      {props.children}
    </button>
  );
}
