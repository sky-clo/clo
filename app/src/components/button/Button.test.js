import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";

it("Button should render a primary button", () => {
  const render = shallow(
    <Button
      onClick={() => null}
      className="test"
      type="submit"
      data-test="my-button"
    >
      My Button
    </Button>
  );
  expect(render).toMatchSnapshot();
});

it("Button should render a secondary button", () => {
  const render = shallow(
    <Button
      onClick={() => null}
      primary={false}
      className="test"
      type="submit"
      data-test="my-button"
    >
      My Button
    </Button>
  );
  expect(render).toMatchSnapshot();
});
