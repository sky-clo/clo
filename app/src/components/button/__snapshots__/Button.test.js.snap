// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a primary button 1`] = `
<button
  className="c-btn c-btn--primary test"
  data-test="my-button"
  onClick={[Function]}
  type="submit"
>
  My Button
</button>
`;

exports[`Button should render a secondary button 1`] = `
<button
  className="c-btn c-btn--secondary test"
  data-test="my-button"
  onClick={[Function]}
  type="submit"
>
  My Button
</button>
`;
